### PR TITLE
OsInterface

### PR DESCRIPTION
**Pull request by RomanLevkovych**
For testing reasons introduced IOsInterface for wrapping syscalls.

### CODE REVIEW

File: applications/ExecutionManager/include/application_handler.hpp [hunk 1]
```diff
@@ -42,7 +43,9 @@ class ApplicationHandler : public IApplicationHandler
             std::vector<std::string>& vectorToConvert) const;
 
 private:
-    const std::string corePath;
+  const std::string corePath;
```
Comment (ihor96berizka) [inline applications/ExecutionManager/include/application_handler.hpp]: m_memberName for all members

File: applications/ExecutionManager/src/os_interface.cpp [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+namespace ExecutionManager
+{
+
+pid_t OsInterface::fork()
+{
+  return ::fork();
+}
+
+int OsInterface::execv(const char *path, char **argv)
```
Comment (ihor96berizka) [inline applications/ExecutionManager/src/os_interface.cpp]: type of second parameter is not as in hpp file.

File: applications/ExecutionManager/include/application_handler.hpp (hunk 1 above)
Comment (RomanLevkovych) [reply applications/ExecutionManager/include/application_handler.hpp]: It is const, so we may leave it without `m_`

Comment (ihor96berizka) [reply applications/ExecutionManager/include/application_handler.hpp]: second member is not const...

Comment (RomanLevkovych) [reply applications/ExecutionManager/include/application_handler.hpp]: ok, it is reasonable

File: applications/ExecutionManager/src/os_interface.cpp [hunk 3]
```diff
@@ -0,0 +1,24 @@
+#include "os_interface.hpp"
+
+#include <unistd.h>
```
Comment (ihor96berizka) [inline applications/ExecutionManager/src/os_interface.cpp]: this include already is included in i_os_interface.hpp file

Comment (rBekesh) [inline applications/ExecutionManager/src/os_interface.cpp]: not needed

**Pull request merged**